### ISSUE TITLE
Can it live side by side with Flask-RestPlus?

### ISSUE THREAD

**Issue by AdilHindistan** [plangrid/flask-rebar#10]
Sorry, this may not be an issue per-se but did not know how else to reach to ask.  In my project, I use Flask-Restplus and Marshmallow. 

Flask-Rebar seems to solve the problem of trying to keep swagger definitions. Looking at the examples, I get the feeling that this is incompatible with Flask-Restplus, but wanted to confirm. Is that the case?
I would love to use them together!

**Comment by barakalon** [comment]
You are correct - they aren't compatible 😕 (at least not in an obvious way that I can see) 

[Flask Restplus's docs](http://flask-restplus.readthedocs.io/en/latest/parsing.html) say `RequestParser` will be replaced with `marshmallow` (or something similar), but I'm not sure of the status on that. Might make sense to borrow some of the [marshmallow to swagger conversion](https://github.com/plangrid/flask-rebar/blob/master/flask_rebar/swagger_generation/marshmallow_to_swagger.py) from here.

**Comment by AdilHindistan** [comment]
OK, thank you! There was a patch to incorporate marshmallow to the project but I have not seen completed since it started in October last year. I do like the fact that it uses classes instead of individual functions and some other pretty cool stuff. It, for example, allows me to define json objects with examples  so both example and model get rendered in swagger. I am guessing model will render fine with rebar but do not see a way to display examples, is that right or did I miss it in documentation?

**Comment by barakalon** [comment]
@AdilHindistan 

> I do like the fact that it uses classes instead of individual functions

Ah, yeah, this should be easy to add as an additional layer to flask rebar. We've actually got some services that do exactly this at PlanGrid. 

I've added an issue for this here: https://github.com/plangrid/flask-rebar/issues/11

>  I am guessing model will render fine with rebar but do not see a way to display examples

You are correct, examples aren't supported yet. Another thing on my todo is to make the swagger generation easily overridable/extendable. 

I've added an issue for this here: https://github.com/plangrid/flask-rebar/issues/12
